scripts/DllManipulatorScript.cs

```diff
@@ -161,15 +157,18 @@ private void OnDisable()
         private void OnDestroy()
         {
             if (_singletonInstance == this)
-            {
-                //Note on threading: Because we don't wait for other threads to finish, we might be stealing function delegates from under their nose if Unity doesn't happen to close them yet.
-                //On Preloaded mode this leads to NullReferenceException, but on Lazy mode the DLL and function would be just reloaded so we would up with loaded DLL after game exit.
-                //Thankfully thread safety with Lazy mode is not implemented yet.
+                Reset();
+        }
 
-                if (DllManipulator.Options != null) // Check that we have initialized
-                    DllManipulator.Reset();
-                _singletonInstance = null;
-            }
+        public void Reset()
```

Comment:
This one doesn't need to be public right?

Reply:
No, its now private